pkg/agent/core/ngt/service/ngt.go

```diff
@@ -235,18 +293,18 @@ func (n *ngt) Start(ctx context.Context) <-chan error {
 			err = nil
 			select {
 			case <-ctx.Done():
-				err = n.CreateIndex(ctx, n.dps)
+				err = n.CreateIndex(ctx, 0)
```

Comment:
I think it would be better to set the field variable here, which enables more flexibility to service package daemon

Reply:
I've just added `defaultPoolSize` constant in service layer.